### PR TITLE
New Resource : `azurerm_api_management_gateway_host_name_configuration` 

### PR DESCRIPTION
**Pull request by sinbai**
Support API Management Gateway Host Name Configuration. API spec is [here](https://github.com/Azure/azure-rest-api-specs/blob/7473936304533e6716fc4563401bf265dda4cb64/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2021-08-01/apimgateways.json#L693).

Test Results:
```
PASS: TestAccApiManagementGatewayHostNameConfiguration_basic (5271.51s)
PASS: TestAccApiManagementGatewayHostNameConfiguration_requiresImport (4982.43s)
PASS: TestAccApiManagementGatewayHostNameConfiguration_complete (4883.23s)
PASS: TestAccApiManagementGatewayHostNameConfiguration_update (5338.44s)
```

### CODE REVIEW

File: internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+				Default:  true,
+				Optional: true,
+			},
+
+			"is_tls10_enabled": {
+				Type:     pluginsdk.TypeBool,
+				Optional: true,
+			},
+
+			"is_tls11_enabled": {
```
Comment (katbyte) [inline internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go:79]: is is redundant when the property ends in enabled, and likely requested to?

Comment (katbyte) [reply internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go:79]:
```suggestion
			"request_client_certificate_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"http2_enabled": {
				Type:     pluginsdk.TypeBool,
				Default:  true,
				Optional: true,
			},

			"tls10_enabled": {
				Type:     pluginsdk.TypeBool,
				Optional: true,
			},

			"tls11_enabled": {
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go (hunk 1 above)
Comment (sinbai) [reply internal/services/apimanagement/api_management_gateway_host_name_configuration_resource.go]: @katbyte thanks for your feedback:). Fixed.

**Review by katbyte — APPROVED** [review]
LGTM 🎲

**Pull request merged**